spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j2/AmqpAppender.java

```diff
@@ -463,25 +530,77 @@ protected AmqpManager(String name) {
 			super(name);
 		}
 
-		private void activateOptions() {
-			this.routingKeyLayout = PatternLayout.createLayout(this.routingKeyPattern
-					.replaceAll("%X\\{applicationId\\}", this.applicationId),
-					null, null, null, Charset.forName(this.charset), false, true, null, null);
-			this.connectionFactory = new CachingConnectionFactory();
-			this.connectionFactory.setHost(this.host);
-			this.connectionFactory.setPort(this.port);
-			if (this.addresses != null) {
-				this.connectionFactory.setAddresses(this.addresses);
+		private boolean activateOptions() {
+			ConnectionFactory rabbitConnectionFactory = createRabbitConnectionFactory();
+			if (rabbitConnectionFactory != null) {
+				this.routingKeyLayout = PatternLayout.createLayout(this.routingKeyPattern
+						.replaceAll("%X\\{applicationId\\}", this.applicationId),
+						null, null, null, Charset.forName(this.charset), false, true, null, null);
+				this.connectionFactory = new CachingConnectionFactory(createRabbitConnectionFactory());
+				if (this.addresses != null) {
+					this.connectionFactory.setAddresses(this.addresses);
+				}
+				if (this.clientConnectionProperties != null) {
+					LogAppenderUtils.updateClientConnectionProperties(this.connectionFactory,
+							this.clientConnectionProperties);
+				}
+				setUpExchangeDeclaration();
+				this.senderPool = Executors.newCachedThreadPool();
+				return true;
 			}
-			this.connectionFactory.setUsername(this.username);
-			this.connectionFactory.setPassword(this.password);
-			this.connectionFactory.setVirtualHost(this.virtualHost);
-			if (this.clientConnectionProperties != null) {
-				LogAppenderUtils.updateClientConnectionProperties(this.connectionFactory,
-						this.clientConnectionProperties);
+			return false;
```

Comment:
This `false` isn't consulted in the `public static AmqpAppender createAppender()`:
```
AmqpAppender appender = new AmqpAppender(name, filter, theLayout, ignoreExceptions, manager);
manager.activateOptions();
appender.startSenders();
```
There we end up with the:
```
Caused by: java.lang.NullPointerException
	at org.springframework.amqp.rabbit.log4j2.AmqpAppender.startSenders(AmqpAppender.java:206)
	at org.springframework.amqp.rabbit.log4j2.AmqpAppender.createAppender(AmqpAppender.java:197)
```
Although I think we really have to return `null` from that `@PluginFactory` instead of uncompleted `AmqpAppender` .